lgc/include/lgc/state/PalMetadata.h

```diff
@@ -155,6 +155,9 @@ class PalMetadata {
   // Updates the PS register information that depends on the exports.
   void updateSpiShaderColFormat(llvm::ArrayRef<ColorExportInfo> exps, bool hasDepthExpFmtZero, bool killEnabled);
 
+  // Sets the finalized 128-bit cache hash.  The version identifies the version of LLPC use to generate the hash.
+  void setFinalized128BitCacheHash(llvm::ArrayRef<uint8_t> finalizedCacheHash, llvm::ArrayRef<uint32_t> version);
```

Comment:
In the previous code review, I heard that we should avoid putting XGL related things in LGC e.g., cache hash, llpc version. It would be better double-checking it before going further (I am worried we realize it is actually not right thing after conducting long code review).

I guess it would be fine because `class PalMetadata` seems to already use note names directly. However, if it is not good, an alternative would be passing general keys and values to `class PalMetadata`.

For example, use `llvm::ArrayRef<StringRef> keys, llvm::ArrayRef<llvm::ArrayRef<uint32_t>> values` for parameters and build the argument in the caller out of LGC.

Reply:
I think that interface would be much more complicated, and make the code harder to read.  It would also force the format of the PAL metadata to be exposed outside of this class, which I would prefer not to do.  

As for the "XGL" being in LGC, the pal metadata is suppose to be more about PAL.  The LGC compiler is targeting PAL, so it has to know about the format of the PAL metadata already.  This class in particular is suppose to be the only place in the LLPC repo that know the format and implementation of the PAL metadata.

I know this is a little inconsistent, but I think this is the correct implementation.